### DISCUSSION TITLE
Only query in-stock products

### DISCUSSION THREAD

**Discussion by stephanboersma** [medusajs/medusa#7163] [Q&A] [answered]
Hi,

How can i query only in-stock products with the medusa-js client? I don't see that kind of filtering capabilities in the `StoreGetProductsParams` type.

I know that I would be able to handle this logic in the frontend but that would make paging quite tricky since there would be a discrepancy between the displayed paged products and the actually queried pages.

This is a requirement because my client has a series of products with few in stock and when they go out of stock they typically do not come back in stock. The number of out of stock products will quickly exceed the number of in-stock products.

Any ideas are welcome :D

Thanks for all the good work on this project - looking forward to 2.0 :)

**Comment by stephanboersma** [comment]
Bump. Still curious if it is possible to filter on inventory_quantity in the products query :)

**Comment by radoslaw-sz** [comment, marked as answer]
hi @stephanboersma  - it is not possible out of the box, but you can do such things:
1) Catch the request via https://docs.medusajs.com/development/api-routes/add-middleware
2) Create your own API endpoint which will execute raw query based on your implementation of inventory_quantity
3) Forward the request caught by 1) and forward it to the 2)

Probably it shall work.

**Reply by stephanboersma** [reply]
Hi @radoslaw-sz,

Thanks, I will try that :)
